src/segments/unity.go

```diff
@@ -0,0 +1,104 @@
+package segments
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/regex"
+)
+
+type Unity struct {
+	props properties.Properties
+	env   platform.Environment
+
+	UnityVersion  string
+	CSharpVersion string
+}
+
+func (u *Unity) Enabled() bool {
+	unityVersion, err := u.GetUnityVersion()
+	if err != nil {
+		u.env.Error(err)
+		return false
+	}
+	if unityVersion == "" {
+		return false
+	}
+	u.UnityVersion = unityVersion
+
+	csharpVersion, err := u.GetCSharpVersion()
+	if err != nil {
+		u.env.Error(err)
+	}
+	u.CSharpVersion = csharpVersion
+
+	return true
+}
+
+func (u *Unity) GetUnityVersion() (version string, err error) {
+	versionFilePath := filepath.Join("ProjectSettings", "ProjectVersion.txt")
+	if !u.env.HasFiles(versionFilePath) {
+		return "", nil
+	}
+
+	versionFileText := u.env.FileContent(versionFilePath)
+	firstLine := strings.Split(versionFileText, "\n")[0]
```

Comment:
Do we want to validate the length before splitting? This will always work, but do we expect more lines?

Reply:
Sure, it wouldn't hurt. From my experience the file always has 2 lines:

```
m_EditorVersion: 2019.4.13f1
m_EditorVersionWithRevision: 2019.4.13f1 (518737b1de84)
```